x/merkledb/db.go

```diff
@@ -749,29 +748,57 @@ func (db *merkleDB) NewIteratorWithStartAndPrefix(start, prefix []byte) database
 // the movement of [node] from [db.nodeCache] to [db.nodeDB] is atomic.
 // As soon as [db.nodeCache] no longer has [node], [db.nodeDB] does.
 // Non-nil error is fatal -- causes [db] to close.
-func (db *merkleDB) onEviction(node *node) error {
-	if node == nil || node.hasValue() {
```

Comment:
I wonder if it makes sense to keep this check as an early return so eviction of non-intermediary nodes remains a no-op and doesn't cause a batch write to disk.

Not sure it really matters, but wanted to point this out.